[bug]: empty status ip in kubevirt cluster causes controller crash

### What happened?

<!-- Try to provide as much information as possible.
If you're reporting a security issue, please check the guidelines for reporting security issues:
https://github.com/kubermatic/kubermatic/blob/main/CONTRIBUTING.md#reporting-a-security-vulnerability -->

Controller crashes repeatedly, when a kubervirt provisioned cluster does not have an adress yet. This causes a disruption in oprations, because the controller is not able to reconcile anymore.

```
{"level":"info","time":"2023-12-18T20:57:26.393Z","logger":"KubeAPIWarningLogger","caller":"rest/warnings.go:144","msg":"instancetype.kubevirt.io/v1alpha1 VirtualMachinePreferences is now deprecated and will be removed in v1."}
{"level":"info","time":"2023-12-18T20:57:26.393Z","caller":"runtime/panic.go:914","msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"kkp-cloud-controller","object":{"name":"ttpfmjb6kz"},"namespace":"","name":"ttpfmjb6kz","reconcileID":"9d51c8fc-47d2-4eba-b41a-5878eeaa43b4"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x329a3ca]

goroutine 865 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x3dbd1c0?, 0x7227ac0?})
	runtime/panic.go:914 +0x21f
k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt.reconcileClusterIsolationNetworkPolicy({0x4d93518, 0xc002414cf0}, 0xc002059000, 0x12?, {0x4dae880, 0xc002806948})
	k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt/network_policy.go:167 +0x4a
k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt.(*kubevirt).reconcileCluster(0xc000a716c8, {0x4d93518, 0xc002414cf0}, 0xc002443000, 0x1?)
	k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt/provider.go:151 +0x27e
k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt.(*kubevirt).ReconcileCluster(0xc0001c6970?, {0x4d93518?, 0xc002414cf0?}, 0x8?, 0x0?)
	k8c.io/kubermatic/v2/pkg/provider/cloud/kubevirt/provider.go:106 +0x1d
k8c.io/kubermatic/v2/pkg/controller/seed-controller-manager/cloud.(*Reconciler).reconcile(0xc0009e79e0, {0x4d93518?, 0xc002414cf0}, 0xc0014100c0, 0xc002443000)
	k8c.io/kubermatic/v2/pkg/controller/seed-controller-manager/cloud/cloud_controller.go:202 +0x88d
k8c.io/kubermatic/v2/pkg/controller/seed-controller-manager/cloud.(*Reconciler).Reconcile.func1()
	k8c.io/kubermatic/v2/pkg/controller/seed-controller-manager/cloud/cloud_controller.go:117 +0x27
k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1/helper.ClusterReconcileWrapper({0x4d93518, 0xc002414cf0}, {0x4dae700, 0xc000516990}, {0x7ffdc87672e8, 0x0}, 0xc002443000, {{0x4d47968, 0x7}, {0x4d47970, ...}, ...}, ...)
	k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1/helper/helper.go:86 +0xd6
k8c.io/kubermatic/v2/pkg/controller/seed-controller-manager/cloud.(*Reconciler).Reconcile(0xc0009e79e0, {0x4d93518, 0xc002414cf0}, {{{0x0?, 0x5?}, {0xc0001c6960?, 0xc001985d48?}}})
	k8c.io/kubermatic/v2/pkg/controller/seed-controller-manager/cloud/cloud_controller.go:109 +0x31d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x4d97e48?, {0x4d93518?, 0xc002414cf0?}, {{{0x0?, 0xb?}, {0xc0001c6960?, 0x0?}}})
	sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002752c0, {0x4d93550, 0xc000aa7e50}, {0x40c9b00?, 0xc001c003c0?})
	sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:316 +0x3c5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002752c0, {0x4d93550, 0xc000aa7e50})
	sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:266 +0x1c9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 262
	sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:223 +0x565
```

This should be handled differently:
https://github.com/kubermatic/kubermatic/blob/56660a79d1e1944efab43f559f0fa26e3b5e422e/pkg/provider/cloud/kubevirt/network_policy.go#L167

Should check, if an IP is set.


### Expected behavior

<!-- What did you expected to happen? -->

The controller should not crash, instead just return an error and reconcile until the address is published.

### How to reproduce the issue?

<!-- Please provide as much information as possible, so we can reproduce the issue on our own. -->

1. Create a new cluster with kubevirt, where the cluster does not get any address information right away. 
2. Controller keeps crashing because of this cluster, disrupting operations.

### How is your environment configured?

- KKP version: v2.24.0
- Shared or separate master/seed clusters?: Single-Stack

### Provide your KKP manifest here (if applicable)

<!-- Providing an applicable manifest (KubermaticConfiguration, Seed, Cluster or other resources) will help us to reproduce the issue.
Please make sure to redact all secrets (e.g. passwords, URLs...)! -->

<details>

```yaml
# paste manifest here
```

</details>

### What cloud provider are you running on?

<!-- AWS, Azure, DigitalOcean, GCP, Hetzner Cloud, Nutanix, OpenStack, Equinix Metal (Packet), VMware vSphere, Other (e.g. baremetal or non-natively supported provider) -->
Kubevirt


### What operating system are you running in your user cluster?

<!-- Ubuntu 20.04, CentOS 7, Rocky Linux 8, Flatcar Linux, ... (optional, bug might not be related to user cluster) -->
Flatcar Linux


### Additional information

<!-- Additional information about the bug you're reporting (optional). -->

--- a/codegen/example-yaml/main.go
+++ b/codegen/example-yaml/main.go
@@ -197,7 +197,7 @@ func createExampleSeed(config *kubermaticv1.KubermaticConfiguration) *kubermatic
 							DNSConfig:                    &corev1.PodDNSConfig{},
 							Images:                       kubermaticv1.KubeVirtImageSources{HTTP: &kubevirtHTTPSource},
 							EnableDefaultNetworkPolicies: ptr.To(true),
-							CustomNetworkPolicies: []*kubermaticv1.CustomNetworkPolicy{
+							CustomNetworkPolicies: []kubermaticv1.CustomNetworkPolicy{
 								{
 									Name: "deny-ingress",
 									Spec: networkingv1.NetworkPolicySpec{

--- a/pkg/apis/kubermatic/v1/datacenter.go
+++ b/pkg/apis/kubermatic/v1/datacenter.go
@@ -740,7 +740,7 @@ type DatacenterSpecKubevirt struct {
 
 	// Optional: CustomNetworkPolicies allows to add some extra custom NetworkPolicies, that are deployed
 	// in the dedicated infra KubeVirt cluster. They are added to the defaults.
-	CustomNetworkPolicies []*CustomNetworkPolicy `json:"customNetworkPolicies,omitempty"`
+	CustomNetworkPolicies []CustomNetworkPolicy `json:"customNetworkPolicies,omitempty"`
 
 	// Images represents standard VM Image sources.
 	Images KubeVirtImageSources `json:"images,omitempty"`

--- a/pkg/apis/kubermatic/v1/zz_generated.deepcopy.go
+++ b/pkg/apis/kubermatic/v1/zz_generated.deepcopy.go
@@ -2181,13 +2181,9 @@ func (in *DatacenterSpecKubevirt) DeepCopyInto(out *DatacenterSpecKubevirt) {
 	}
 	if in.CustomNetworkPolicies != nil {
 		in, out := &in.CustomNetworkPolicies, &out.CustomNetworkPolicies
-		*out = make([]*CustomNetworkPolicy, len(*in))
+		*out = make([]CustomNetworkPolicy, len(*in))
 		for i := range *in {
-			if (*in)[i] != nil {
-				in, out := &(*in)[i], &(*out)[i]
-				*out = new(CustomNetworkPolicy)
-				(*in).DeepCopyInto(*out)
-			}
+			(*in)[i].DeepCopyInto(&(*out)[i])
 		}
 	}
 	in.Images.DeepCopyInto(&out.Images)

--- a/pkg/provider/cloud/kubevirt/network_policy.go
+++ b/pkg/provider/cloud/kubevirt/network_policy.go
@@ -152,7 +152,7 @@ func clusterImporterNetworkPolicyReconciler() reconciling.NamedNetworkPolicyReco
 	}
 }
 
-func customNetworkPolicyReconciler(existing *kubermaticv1.CustomNetworkPolicy) reconciling.NamedNetworkPolicyReconcilerFactory {
+func customNetworkPolicyReconciler(existing kubermaticv1.CustomNetworkPolicy) reconciling.NamedNetworkPolicyReconcilerFactory {
 	return func() (string, reconciling.NetworkPolicyReconciler) {
 		return existing.Name, func(np *networkingv1.NetworkPolicy) (*networkingv1.NetworkPolicy, error) {
 			np.Name = existing.Name
@@ -163,8 +163,13 @@ func customNetworkPolicyReconciler(existing *kubermaticv1.CustomNetworkPolicy) r
 }
 
 func reconcileClusterIsolationNetworkPolicy(ctx context.Context, cluster *kubermaticv1.Cluster, dc *kubermaticv1.DatacenterSpecKubevirt, client ctrlruntimeclient.Client) error {
+	var nameservers []string
+	if dc.DNSConfig != nil {
+		nameservers = dc.DNSConfig.Nameservers
+	}
+
 	namedNetworkPolicyReconcilerFactories := []reconciling.NamedNetworkPolicyReconcilerFactory{
-		clusterIsolationNetworkPolicyReconciler(cluster.Status.Address.IP, dc.DNSConfig.Nameservers),
+		clusterIsolationNetworkPolicyReconciler(cluster.Status.Address.IP, nameservers),
 		clusterImporterNetworkPolicyReconciler(),
 	}
 	if err := reconciling.ReconcileNetworkPolicies(ctx, namedNetworkPolicyReconcilerFactories, cluster.Status.NamespaceName, client); err != nil {

